deeppavlov/models/kbqa/query_generator.py

```diff
@@ -0,0 +1,597 @@
+            entities_to_leave: how many entities to leave after entity linking
+            rels_to_leave: how many relations to leave after relation ranking
+            return_answers: whether to return answers or candidate answers
+            debug: whether to print debug information
+            **kwargs:
+        """
+        super().__init__(save_path=None, load_path=load_path)
+        self.template_matcher = template_matcher
+        self.linker = linker
+        self.wiki_parser = wiki_parser
+        self.rel_ranker = rel_ranker
+        self.rank_rels_filename_1 = rank_rels_filename_1
+        self.rank_rels_filename_2 = rank_rels_filename_2
+        self.entities_to_leave = entities_to_leave
+        self.rels_to_leave = rels_to_leave
+        self.return_answers = return_answers
+        self.debug = debug
+        self.load()
+
+    def load(self) -> None:
+        with open(self.load_path / self.rank_rels_filename_1, 'r') as fl1:
+            lines = fl1.readlines()
+            self.rank_list_0 = [line.split('\t')[0] for line in lines]
+
+        with open(self.load_path / self.rank_rels_filename_2, 'r') as fl2:
+            lines = fl2.readlines()
+            self.rank_list_1 = [line.split('\t')[0] for line in lines]
+
+    def save(self) -> None:
+        pass
+
+    def __call__(self, question_tuple: List[str],
+                 template_type: List[str],
+                 entities_from_ner: List[str]) -> List[Tuple[str]]:
+
+        candidate_outputs = []
+        question = question_tuple[0]
+        self.template_num = int(template_type[0])
+
+        question = question.replace('"', "'").replace('{', '').replace('}', '').replace('  ', ' ')
+        entities_from_template, rels_from_template, query_type_template = self.template_matcher(question)
+        if query_type_template == "simple":
+            self.template_num = 7
+
+        if self.debug:
+            log.debug(f"question: {question}\n")
+            log.debug(f"template_type {self.template_num}")
+
+        if entities_from_template:
+            entity_ids = self.get_entity_ids(entities_from_template)
+            if self.debug:
+                log.debug(f"entities_from_template {entities_from_template}")
+                log.debug(f"rels_from_template {rels_from_template}")
+                log.debug(f"entity_ids {entity_ids}")
+
+            candidate_outputs = self.find_candidate_answers(question, entity_ids, rels_from_template)
+
+        if not candidate_outputs and entities_from_ner:
+            entity_ids = self.get_entity_ids(entities_from_ner)
+            log.debug(f"entities_from_ner {entities_from_ner}")
+            log.debug(f"entity_ids {entity_ids}")
+            self.template_num = int(template_type[0])
+            candidate_outputs = self.find_candidate_answers(question, entity_ids, rels_from_template=None)
+
+        if self.return_answers:
+            answer = self.rel_ranker([question], candidate_outputs, True)
```

Comment:
`RelRankerInfer.__call__` method receives only two arguments, `True` is unexpected.

Reply:
I forgot add to typing RelRankerBertInfer. self.rel_ranker can be either RelRankerInfer or RelRankerBertInfer, depending on the config.